### PR TITLE
feat: Favourite Investor API endpoint

### PR DESCRIPTION
**Pull request by martintomas**
New API endpoint which allows to mark/unmark investors as favourite. Controller is basically same as ones used for marking Project or Project Developers as favourite.

I have moved all policy checks to cancancan for these controllers, therefore You are now required to have approved account otherwise you receive forbidden error.

## Testing instructions

New API endpoint for making Investors favourite is available via Rswag documentation, therefore everything can be tested via there. Please make sure to use approved account, otherwise you should receive forbidden error.

## Tracking

https://vizzuality.atlassian.net/browse/LET-526
https://vizzuality.atlassian.net/browse/LET-527
https://vizzuality.atlassian.net/browse/LET-554

### CODE REVIEW

File: backend/spec/requests/api/v1/favourite_investors_spec.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      security [csrf: [], cookie_auth: []]
+      parameter name: :investor_id, in: :path, type: :string
+      parameter name: :empty, in: :body, schema: {type: :object}, required: false
+
+      let(:investor) { create :investor }
+      let(:investor_id) { investor.id }
+      let(:user) { create :user, account: create(:account, :approved) }
+
+      it_behaves_like "with not authorized error", csrf: true
+      it_behaves_like "with forbidden error", csrf: true
```
Comment (tsubik) [inline backend/spec/requests/api/v1/favourite_investors_spec.rb:18]: Maybe we need to refactor specs to make those behaviors more clear. Like `it_behaves_like "with forbidden error", user: -> { create(:user) }`. To know for what kind of user we are testing.

Comment (martintomas) [reply backend/spec/requests/api/v1/favourite_investors_spec.rb:18]: Like it, will do it :)

File: backend/spec/support/shared_examples/api/with_forbidden_error.rb [hunk 2]
```diff
@@ -1,15 +1,11 @@
...
   response "403", "Forbidden" do
     let("X-CSRF-TOKEN") { get_csrf_token } if csrf
 
     before do
-      if require_project_developer
-        sign_in create(:user_project_developer)
-      else
-        sign_in create(:user)
-      end
+      sign_in instance_exec(&user) if user.present?
```
Comment (tsubik) [inline backend/spec/support/shared_examples/api/with_forbidden_error.rb:8]: Nice :+1:

**Review by tsubik — APPROVED** [review]
:rocket:

**Pull request merged**